src/components/MailingList/MailingList.jsx

```diff
@@ -30,51 +34,106 @@ const EmailSubmitButton = styled.input`
     border-top-right-radius: 10px;
     border-bottom-right-radius: 10px;
     &:hover {
-      background: rgb(222, 103, 63);
+        background: rgb(222, 103, 63);
+    }
+    &:disabled {
+        background: #aaa;
+    }
+    &:disabled:hover {
+        cursor: not-allowed;
     }
 `;
 
-export default function MailingList() {
-  const [address, setAddress] = useState("");
-  const [result, setResult] = useState(null);
+const LoadingAnimation = styled.span`
+    ${(props) => (props.active ? "animation: spin 2s linear infinite;" : "")}
+    @keyframes spin {
+        0% {
+            transform: rotate(0deg);
+        }
+        100% {
+            transform: rotate(360deg);
+        }
+    }
+    display: flex;
+    justify-content: center;
+    align-items: center;
+`;
 
-  async function addEmailToList(e) {
-    e.preventDefault();
-    setResult(null);
+const READY_STATUS = "ready";
+const LOADING_STATUS = "loading";
+const SUCCESS_STATUS = "success";
+const FAILURE_STATUS = "failure";
 
-    // make email post request
-    const response = await fetch('https://api.michhackers.com/email/add', {
-      method: 'POST',
-      headers: {
-        "Content-Type": "application/json"
-      },
-      body: JSON.stringify({
-        email: address
-      })
-    });
+const statusToTitle = {};
+statusToTitle[READY_STATUS] = "Send";
```

Comment:
I think this part would be better if it was an object like {title: "Send", icon: "paper-plane"} to avoid that nasty nested ternary. Alternatively, move the setting of the icon to a variable and use `switch` instead.